cmd/virt-tail/main.go

```diff
@@ -0,0 +1,229 @@
+			}
+		} else {
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+
+	// initial timeout for serial console socket creation
+	socketCheckCh := make(chan int)
+	time.AfterFunc(initialSocketTimeout, func() {
+		socketCheckCh <- 1
+	})
+
+	// Start listening for events.
+	for {
+		select {
+		case <-socketCheckCh:
+			if !checkFile(socketFile, socketExists) {
+				rerr := errors.New("socketFile is still not ready")
+				log.Log.V(3).Infof("watchFS error: %v", rerr)
+				return rerr
+			} else {
+				socketExists = true
+			}
+			if checkFile(termFile, false) {
+				log.Log.V(3).Infof("watchFS error: termFile was already there")
+				return &TermFileError{}
+			}
+		case event := <-watcher.Events:
+			if event.Has(fsnotify.Create) {
+				if event.Name == termFile {
+					// termination file got created, we should quickly terminate
+					terr := &TermFileError{}
+					log.Log.V(3).Infof("watchFS error: %v", terr)
+					return terr
+				} else if event.Name == socketFile {
+					// socket file got created
+					socketExists = true
+				}
+			} else if event.Has(fsnotify.Remove) {
+				if event.Name == socketFile {
+					// socket file got deleted, we should quickly terminate
+					rerr := errors.New("socketFile got removed")
+					log.Log.V(3).Infof("watchFS error: %v", rerr)
+					return rerr
+				}
+			}
+		case werr := <-watcher.Errors:
+			log.Log.V(3).Infof("watcher error: %v", werr)
+			return werr
+		case <-gctx.Done():
+			return gctx.Err()
+		}
+	}
+}
+
+func main() {
+	// set new default verbosity, was set to 0 by glog
+	goflag.Set("v", "2")
+	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.CommandLine.ParseErrorsWhitelist = pflag.ParseErrorsWhitelist{UnknownFlags: true}
+	pflag.Parse()
+
+	log.InitializeLogging("virt-tail")
+
+	// check if virt-launcher verbosity should be changed
+	if verbosityStr, ok := os.LookupEnv("VIRT_LAUNCHER_LOG_VERBOSITY"); ok {
+		if verbosity, err := strconv.Atoi(verbosityStr); err == nil {
+			log.Log.SetVerbosityLevel(verbosity)
+			log.Log.V(3).Infof("set log verbosity to %d", verbosity)
+		} else {
+			log.Log.Warningf("failed to set log verbosity. The value of logVerbosity label should be an integer, got %s instead.", verbosityStr)
+		}
+	}
+
+	logFile := pflag.String("logfile", "", "path of the logfile to be streamed")
+
+	pflag.Parse()
+
+	if logFile == nil || *logFile == "" {
+		log.Log.V(3).Infof("logfile flags must be provided")
+		os.Exit(1)
+	}
+	gLogFile = *logFile
+
+	// Create context that listens for the interrupt signal from the container runtime.
+	ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt)
+
+	g, gctx = errgroup.WithContext(ctx)
+	g.Go(tailLogs)
+	g.Go(watchFS)
+
+	// wait for all errgroup goroutines
+	err := g.Wait()
+	if err != nil {
+		if errors.Is(err, context.Canceled) || errors.Is(err, &TermFileError{}) {
+			os.Exit(0)
+		} else {
+			log.Log.V(3).Infof("received error: %v", err)
+			os.Exit(1)
+		}
+	} else {
+		// clean
+		os.Exit(0)
+	}
```

Comment:
```suggestion
	if err := g.Wait(); err != nil && !errors.Is(err, context.Canceled) && !errors.Is(err, &TermFileError{}) {
		os.Exit(1)
	}
```

This should suffice?

Reply:
done